### ISSUE TITLE
Grouping agents Cleaner

### ISSUE THREAD

**Issue by rejkid** [JamesBremner/Agents2Tasks#22]
One more requirement came up which I think would be hard to implement in an algorithm. Every agent (task Cleaner) has been assigned one letter ("A" or "B" or "C" or ... etc ) for grouping purposes.
When agents for the time slot are assigned, they should come from the same group. So for example :

a iwhite@gmail.com&1961-07-02&C 1 Cleaner 
a iblack@gmail.com&1961-07-02&B 1 Cleaner 
a awhite@gmail.com&1998-06-07&C 1 Cleaner 
a ablack@gmail.com&1998-06-07&B 1 Cleaner 
a asmith@gmail.com&1998-06-07&A 1 Cleaner

t 202310220830 Cleaner  Cleaner
t 202310221000 Cleaner  Cleaner
t 202310291000 Cleaner 

 ... then for "t 202310220830" there can be :
 a iwhite@gmail.com&1961-07-02&C 1 Cleaner 
 a awhite@gmail.com&1998-06-07&C 1 Cleaner  

... and for the next time slot "t 202310221000" there can be:
a iblack@gmail.com&1961-07-02&B 1 Cleaner 
a ablack@gmail.com&1998-06-07&B 1 Cleaner 

... and for the next time slot  "t 202310291000" there can be
a asmith@gmail.com&1998-06-07&A 1 Cleaner

If you also think its to difficult I will tell my client that it will have to be done manually using my GUI.

**Comment by JamesBremner** [comment]
This appears to be exactly the same as grouping by family ( #21 ).  Whether you call the 'family group' by the name "Smith" or "A" make no difference to the code.  It will attempt to assign tasks to agents that share the same 'family group name"

This issue should be closed as a duplicate of #21

**Comment by rejkid** [comment]
Can two grouping fields coexist (one grouping by family and one for Grouping cleaners)?

**Comment by JamesBremner** [comment]
I do not understand.  What does "for Grouping cleaners" mean?

It would help if you used the format described in https://github.com/JamesBremner/Agents2Tasks/wiki/File-format for your example.

`a iwhite@gmail.com 1 C cleaner`

specifies

```
name: iwhite@gmail.com
cost: 1
group: C
can do tasks: cleaner
```

while your example

`a iwhite@gmail.com&1961-07-02&C 1 Cleaner`

specifies 

```
name: iwhite@gmail.com&1961-07-02&C
cost: 1
group: Cleaner
can do tasks: none
```

**Comment by rejkid** [comment]
Would that work?

For Cleaners, line would look like:
a iwhite@gmail.com&1961-07-02 1 iwhite@gmail.com A Cleaner

specifies

name: iwhite@gmail.com&1961-07-02
cost: 1
family: iwhite@gmail.com
group: A
can do tasks: Cleaner


For everybody else (non Cleaners)
a ismith@gmail.com&1961-07-02 1 ismith@gmail.com Teacher

specifies

name: ismith@gmail.com&1961-07-02
cost: 1
family: ismith@gmail.com
group: None
can do tasks: Teacher

**Comment by JamesBremner** [comment]
Every line must have the same number of fields.

`a iwhite@gmail.com&1961-07-02 1 iwhite@gmail.com A Cleaner` has 6 fields


`a ismith@gmail.com&1961-07-02 1 ismith@gmail.com Teacher` has 5 fields

The result will be horrible confusion

**Comment by rejkid** [comment]
Hi James. I think it will be a very complicated code on your side to do assignment with two grouping fields 
1) for families
and
2) for Cleaners

... even if we come up with some solution to have the same number of fields (e.g. "None" field for agents that are not cleaners) 

Also for agents that can be both Cleaners and non cleaners how are you going to handle assignment for that - I think it is to complex
If you agree (you might think about some way to do the assignment algorithm) I will get back to my client and tell them that assignment for Cleaners will have to be  a manual process through my GUI.
It all get complicated because people that are cleaners want to work in their teams (group A B C etc) but they are not families.

**Comment by JamesBremner** [comment]
I do not understand what you want to do with the "cleaner group"  How does it differ from any other group such A, or B or C?

If you want some agents to be in a group called "cleaner" just specify them in that group. I do not understand why you need another field.

**Comment by rejkid** [comment]
If you want some agents to be in a group called "cleaner" just specify them in that group. I do not understand why you need another field.
   Do you mean use family group to specify the cleaner's group?

**Comment by JamesBremner** [comment]
Yes.

**Comment by rejkid** [comment]
What if we specify mixed mode, by that I mean when we specify in the same "a" record Cleaner and non Cleaner tasks - how family group field would be interpreted - as a family or Cleaner group

**Comment by JamesBremner** [comment]
The family group is interpreted the same, no matter what tasks the agent can do.  If the slot has a member of the same family group assigned, then the agent will be preferred for the other tasks in the slot.

**Comment by JamesBremner** [comment]
This should be closed as a duplicate of #21

**Comment by rejkid** [comment]
Duplicate of #21
